src/Adapter/Tornado/EventLoop.php

```diff
@@ -231,22 +277,32 @@ public function idle(): Promise
      */
     public function delay(int $milliseconds): Promise
     {
+        /** @var PendingPromise $promise */
+        $promise = null;
         $endTime = microtime(true) + $milliseconds / 1000 /* milliseconds in 1 second */;
 
-        return $this->async((function () use ($endTime): \Generator {
+        $promise = $this->async((function () use ($endTime, &$promise): \Generator {
             while (microtime(true) < $endTime) {
-                yield $this->idle();
+                try {
+                    yield $this->idle();
+                } catch (\Throwable $throwable) {
```

Comment:
I think you should catch only `CancelledException`, and cancel the promise… not even sure if it's required or not… 🤔